frontend/src/common/types/common.ts

```diff
@@ -129,7 +115,21 @@ export type PageMetadataInResponse = {
    * Is there a next page?
    */
   hasNextPage: boolean;
-} & PaginationOptions;
+};
+
+/**
+ * A generic paginated response structure for all the paginated responses from APIs.
+ */
+export interface PaginatedResponse<T> {
```

Comment:
I think this is the one, except I changed it from using a 'type' to 'interface', that way more specific types can extend it with the 'extends' keyword (instead of using & for types).